yt/yt/core/misc/async_slru_cache-inl.h

```diff
@@ -1323,6 +1431,14 @@ bool TAsyncSlruCacheBase<TKey, TValue, THash>::TInsertCookie::IsActive() const
     return Active_;
 }
 
+template <class TKey, class TValue, class THash>
+void TAsyncSlruCacheBase<TKey, TValue, THash>::TInsertCookie::UpdateWeight(i64 cookieWeight)
+{
+    if (Active_) {
```

Comment:
Is it OK to invoke this for non-active cookies at all?

Reply:
It seems should be ok to cancel cookie at any time.
Weight updates for cancelled cookie must be simply ignored.

In this case UpdateCookieWeight should handle missing item in the map (to close race with cancel).